src/utils/useCanPublish.ts

```diff
@@ -0,0 +1,55 @@
+import { useContext, useEffect, useRef, useState } from 'react';
+import { shallowEqual, useDispatch, useSelector } from 'react-redux';
+import { useGlobal } from 'reactn';
+import { IProfileStrings, IState } from '../model';
+import * as action from '../store';
+import { TokenContext } from '../context/TokenProvider';
+import { profileSelector } from '../selector';
+
+export const useCanPublish = () => {
+  const [canPublish, setCanPublish] = useState(false);
+  const askingRef = useRef(false);
+  const [isOffline] = useGlobal('offline');
+  const { accessToken } = useContext(TokenContext).state;
+  const [errorReporter] = useGlobal('errorReporter');
+  const paratext_canPublish = useSelector(
+    (state: IState) => state.paratext.canPublish
+  );
+  const paratext_canPublishStatus = useSelector(
+    (state: IState) => state.paratext.canPublishStatus
+  );
+  const dispatch = useDispatch();
+  const getCanPublish = (token: string, errorReporter: any, message: string) =>
+    dispatch(action.getCanPublish(token, errorReporter, message));
+  const resetCanPublish = () => dispatch(action.resetCanPublish());
+  const t: IProfileStrings = useSelector(profileSelector, shallowEqual);
+
+  useEffect(() => {
+    if (!isOffline) {
+      if (!askingRef.current && accessToken && !paratext_canPublishStatus) {
+        askingRef.current = true; //so we only call it once
```

Comment:
Where do we reset this askingRef?

Reply:
why would we need to call it again?

Reply:
perhaps only if it failed because of a bad connection?